src/database.cpp

```diff
@@ -1,11 +1,23 @@
 #include "database.hpp"
 
-void Database::compact()
+Database::Database()
+{
+    createRTree = false;
+}
+Database::Database(bool _createRTree) : createRTree(_createRTree)
 {
-    rtree =
+}
+
+void Database::build_rtree()
```

Comment:
I think this function can be marked `const` (also in header) because it does not modify the Database classes member attributes

Reply:
See below, can't be marked as const :(